In [1]:
from pudu.apis.foxx_api import *
from pudu.apis.pudu_api import *


In [ ]:

start_time = "2025-08-12 00:00:00"
end_time = "2025-08-15 23:59:59"
charging_table = get_battery_health_list(start_time, end_time, shop_id='533370001')

In [1]:
import time
int(time.time())

1755387517

In [6]:
get_list_stores()['list']

[{'company_id': '100004935',
  'company_name': 'Foxx Development Inc.',
  'shop_id': '434100007',
  'shop_name': 'USF demo'},
 {'company_id': '100004935',
  'company_name': 'Foxx Development Inc.',
  'shop_id': '434300005',
  'shop_name': 'USF'},
 {'company_id': '100004935',
  'company_name': 'Foxx Development Inc.',
  'shop_id': '450270000',
  'shop_name': 'salt-lake-airport'},
 {'company_id': '100004935',
  'company_name': 'Foxx Development Inc.',
  'shop_id': '513370000',
  'shop_name': 'Demo'},
 {'company_id': '100004935',
  'company_name': 'Foxx Development Inc.',
  'shop_id': '523670000',
  'shop_name': 'demo-all'},
 {'company_id': '100004935',
  'company_name': 'Foxx Development Inc.',
  'shop_id': '530470002',
  'shop_name': 'Mobiltech'},
 {'company_id': '100004935',
  'company_name': 'Foxx Development Inc.',
  'shop_id': '533370001',
  'shop_name': 'University of Florida'}]

In [8]:
get_list_robots(shop_id='533370001')

{'count': 2,
 'list': [{'mac': 'AC:D9:29:88:F8:BF',
   'shop_id': '533370001',
   'shop_name': 'University of Florida',
   'sn': '811135422060228'},
  {'mac': 'AC:D9:29:88:F8:37',
   'shop_id': '533370001',
   'shop_name': 'University of Florida',
   'sn': '811135422060216'}]}

In [12]:
get_cleaning_report_list(start_time='2025-08-15 00:00:00', end_time='2025-08-15 23:59:59', shop_id='533370001', sn='811135422060216')

{'limit': 100,
 'list': [{'clean_area': 803.3912353515625,
   'clean_time': 4743,
   'create_time': '2025-08-15 23:47:14',
   'end_time': 1755301629,
   'mac': 'AC:D9:29:88:F8:37',
   'mode': 1,
   'report_id': '1153977308077101056',
   'sn': '811135422060216',
   'start_time': 1755296095,
   'status': 4,
   'sub_mode': 0,
   'task_area': 783.78,
   'task_name': 'Bld_205_1st_Task ALL'},
  {'clean_area': 160.90219116210938,
   'clean_time': 955,
   'create_time': '2025-08-15 18:25:50',
   'end_time': 1755282346,
   'mac': 'AC:D9:29:88:F8:37',
   'mode': 1,
   'report_id': '1153914846480961536',
   'sn': '811135422060216',
   'start_time': 1755281203,
   'status': 4,
   'sub_mode': 0,
   'task_area': 102.3175,
   'task_name': 'ground_elevator_new'},
  {'clean_area': 45.36689376831055,
   'clean_time': 281,
   'create_time': '2025-08-15 17:57:19',
   'end_time': 1755280635,
   'mac': 'AC:D9:29:88:F8:37',
   'mode': 1,
   'report_id': '1153909701508599808',
   'sn': '811135422060216',
   '

curl -X POST "http://18.222.168.242:8000/api/pudu/webhook" -H "Content-Type: application/json" -H "CallbackCode: 1vQ6MfUxqyoGMRQ9nK8C4pSkg1Qsa3Vpq" -d '{"callback_type": "robotStatus", "data": {"sn": "811135422060216", "timestamp": 1755387517, "run_status": "Online"}}'

curl -X POST "http://44.193.26.65:8000/api/pudu/webhook" -H "Content-Type: application/json" -H "CallbackCode: 1vQ6MfUxqyoGMRQ9nK8C4pSkg1Qsa3Vpq" -d '{"callback_type": "robotErrorWarning", "data": {"sn": "811064412050012", "timestamp": 1755387517, "error_type": "LostLocalization", "error_level": "Warning", "error_detail": "",  "error_id": "test00003"}}'

curl -X GET "http://18.222.168.242:8000/api/pudu/webhook/health"

curl -X GET "http://44.193.26.65:8000/api/pudu/webhook/health"

import time

In [7]:
for result in charging_table['list']:
    print(result['sn'], result['work_status'])

811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060216 11
811135422060228 0
811135422060216 11
811135422060216 11
811135422060216 11
8111354220602

In [2]:
sns = set()
for result in charging_table['list']:
    sns.add(result['sn'])
sns

{'811135422060216', '811135422060228'}

In [3]:
battery_health_dict = {}
sn = set()
for health_record in charging_table['list']:
    key = health_record['sn']
    sn.add(key)
    battery_health_dict[key] = {
        'cycle': health_record.get('cycle', None),
        'soc': health_record.get('soc', None),
        'soh': health_record.get('soh', None)
    }

In [4]:
get_list_robots(shop_id='533370001')['list']

[{'mac': 'AC:D9:29:88:F8:BF',
  'shop_id': '533370001',
  'shop_name': 'University of Florida',
  'sn': '811135422060228'},
 {'mac': 'AC:D9:29:88:F8:37',
  'shop_id': '533370001',
  'shop_name': 'University of Florida',
  'sn': '811135422060216'}]

In [7]:
sn

{'811135422060216', '811135422060228'}

In [9]:
results = get_charging_record_list(start_time, end_time, shop_id='533370001', timezone_offset=0)['list']

In [15]:
sns = set()
for result in results:
    sns.add(result['sn'])
sns

{'811064412050012', '811135422060216', '811135422060228'}

In [2]:
s = get_charging_table(start_time, end_time, location_id='533370001')
s

Robot Name         Robot SN          Start Time  \
0   Building_205_Dental_Science  811135422060216 2025-08-15 22:14:15   
1   Building_205_Dental_Science  811135422060216 2025-08-15 20:19:49   
2   Building_205_Dental_Science  811135422060216 2025-08-15 20:18:45   
3   Building_43_Marston_Library  811135422060228 2025-08-15 00:15:49   
4                     uf-backup  811064412050012 2025-08-14 21:55:31   
5   Building_205_Dental_Science  811135422060216 2025-08-14 16:12:11   
6   Building_205_Dental_Science  811135422060216 2025-08-14 15:36:50   
7   Building_205_Dental_Science  811135422060216 2025-08-14 14:21:02   
8   Building_205_Dental_Science  811135422060216 2025-08-13 21:43:40   
9                     uf-backup  811064412050012 2025-08-13 21:29:11   
10  Building_43_Marston_Library  811135422060228 2025-08-13 20:07:33   
11  Building_43_Marston_Library  811135422060228 2025-08-13 19:56:43   
12  Building_205_Dental_Science  811135422060216 2025-08-13 13:18:08   
13  Building_205_Dental_Science  811135422060216 2025-08-13 12:56:45   
14  Building_43_Marston_Library  811135422060228 2025-08-13 12:46:48   
15  Building_43_Marston_Library  811135422060228 2025-08-13 12:46:48   
16  Building_43_Marston_Library  811135422060228 2025-08-13 12:46:30   
17  Building_43_Marston_Library  811135422060228 2025-08-13 12:46:30   

              End Time  Duration Initial Power Final Power Power Gain Status  
0  2025-08-16 00:01:22  1h 47min           94%        100%        +6%   Done  
1  2025-08-15 20:20:52  0h 01min           94%         94%        +0%   Done  
2  2025-08-15 21:51:07  1h 32min           51%         94%       +43%   Done  
3  2025-08-15 03:11:19  2h 55min           31%        100%       +69%   Done  
4  2025-08-14 22:00:01  0h 04min           11%         14%        +3%   Done  
5  2025-08-14 16:43:19  0h 31min           94%        100%        +6%   Done  
6  2025-08-14 15:36:50  0h 00min           94%         94%        +0%   Done  
7  2025-08-14 14:21:02  0h 00min          100%        100%        +0%   Done  
8  2025-08-13 21:47:40  0h 04min           34%         35%        +1%   Done  
9  2025-08-13 21:54:41  0h 25min           52%         66%       +14%   Done  
10 2025-08-13 20:16:47  0h 09min           68%         73%        +5%   Done  
11 2025-08-13 21:56:03  1h 59min           10%         69%       +59%   Done  
12 2025-08-13 13:27:32  0h 09min           95%        100%        +5%   Done  
13 2025-08-13 13:10:22  0h 13min           89%         95%        +6%   Done  
14 2025-08-13 12:46:54  0h 00min           73%         73%        +0%   Done  
15 2025-08-13 12:46:54  0h 00min           73%         73%        +0%   Done  
16 2025-08-13 12:48:59  0h 02min           72%         73%        +1%   Done  
17 2025-08-13 12:48:59  0h 02min           72%         73%        +1%   Done